packages/integrations/cloudflare/README.md

```diff
@@ -92,16 +92,17 @@ To do this:
 
 ## Environment Variables
 
-As Cloudflare Pages Functions [provides environment variables differently](https://developers.cloudflare.com/pages/platform/functions/#adding-environment-variables-locally), private environment variables needs to be set through [`vite.define`](https://vitejs.dev/config/shared-options.html#define) to work in builds.
+As Cloudflare Pages Functions [provides environment variables per request](https://developers.cloudflare.com/pages/platform/functions/#adding-environment-variables-locally), private environment variables should only be accessed when a request has happened. Usually, this means moving environment variable access inside a function.
 
 ```js
-// astro.config.mjs
-export default {
-  vite: {
-    define: {
-      'process.env.MY_SECRET': JSON.stringify(process.env.MY_SECRET),
-    },
-  },
+// util.js
+
+// ❌ Do not access environment variables on initialization
+const serverUrl = import.meta.env.SERVER_URL;
+
+export function fetchData() {
+  // ✅ Access inside a function instead
+  return fetch(import.meta.env.SERVER_URL + "/api/data");
```

Comment:
I learnt the "no wrong code" rule from y'all too, but I can't quite think of an alternative to show what not to do. Maybe it would be clearer if this shows an endpoint and have `const serverUrl = import.meta.env.SERVER_URL;` inside the function with extra comments. Will tweak this!

Reply:
Added it in [43303e3](https://github.com/withastro/astro/pull/5301/commits/43303e36990d4261bda67b6bc88ed3bbb6cb6214). It doesn't cover for cases where an Astro file imports a `util.js` that does fetch calls though like the previous example, but it feels like the endpoint case would be more common.